### ISSUE TITLE
[2.0.5] Explicit `false` not working (`options.sourceMap`)

### ISSUE THREAD

**Issue by leebenson** [postcss/postcss-loader#250]
I have a function that dynamically sets whether sourceMaps should be included (amongst other options):

```js
getExtractCSSLoaders(extractCSS, sourceMap = false) {
  return (function* loadCss() {
    for (const loader of css.loaders) {
      // Iterate over CSS/SASS/LESS and yield local and global mod configs
      for (const mod of css.getModuleRegExp(loader.ext)) {
        yield {
          test: new RegExp(mod[0]),
          loader: extractCSS.extract({
            use: [
              {
                loader: 'css-loader',
                query: Object.assign({}, css.loaderDefaults, {
                  sourceMap,
                }, mod[1]),
              },
              // 'postcss-loader',
              {
                loader: 'postcss-loader',
                options: {
                  sourceMap: false,
                },
              },
              ...loader.use,
            ],
            fallback: 'style-loader',
          }),
        };
      }
    }
  }());
},
```

Explicitly setting `sourceMap: false` seems to kill/bypass postcss:

```js
// Explicitly setting `sourceMap` to the default `false` kills it...
{
  loader: 'postcss-loader',
  options: {
    sourceMap: false,
  },
},
```

Setting it to true works fine:

```js
{
  loader: 'postcss-loader',
  options: {
    sourceMap: false,
  },
},
```

And ignoring the option is also fine:

```js
{
  loader: 'postcss-loader',
},
```

A temporary workaround is to do a slightly uglier:

```js
{
  // ...
  sourceMap ? {
    loader: 'postcss-loader',
    options: {
      sourceMap: true,
    },
  } : 'postcss-loader',
  // ...
}
```

**Comment by michael-ciniawsky** [comment]
Yep, it's a know issue with the current config algo using `!sourceMap` internally, I try to fix it as soon as possible, but please omit it for now

**Comment by michael-ciniawsky** [comment]
Fixed && Released in [`2.0.6`](https://github.com/postcss/postcss-loader/releases/tag/v2.0.6) please test and verify :)

**Comment by leebenson** [comment]
Works great, thanks @michael-ciniawsky !
